enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,512 @@
+
+**OperatorHub and Console changes**: Allow for import of additional, well-known, bundle annotations which will result 
+in ENV variables on the Subscription object. Show operator is enabled for Token-based use based on Console determining 
+cluster is running in token-based authentication enabled cluster by reading the `.spec.serviceAccountIssuer` from the 
+Authentication CR, `.spec.platformSpec.type` from the Infrastructure CR,`.spec.credentialsMode` from the 
+CloudCredentials CR  AND bundle has token-based auth enabled annotation. This info should be written to a new field on 
+Subscription objects. The Subscription fields will allow UX for the information needed by CCO or webhook, for input of 
+the cloud credentials. Generate a manual ack for operator upgrade when cloud resources are involved. This can be 
+determined by parsing the Subscription objects. Ensures admin signs off that cloud resources are provisioned and working
+as intended before an operator upgrade.
+
+**Operator team/ Operator SDK changes**: Follow new guidelines for allowing for the operator to work on token auth 
+enabled cluster. New guidelines would include the following to use CCO Token mode:
+
+- changing CRD to add CredentialRequest references and putting those referenced
+- CredentialRequests into a defined directory in the bundle
+- Add a bundle annotation claiming token-based authentication support
+- Add a bundle annotation indicating which permissions are required
+- add the projected ServiceAccount volume to deployment;
+- handle the Secret themselves (like today’s CCO Mint mode, read and use the credentials in the Secret whose name they 
+  know from the CredentialsRequest and is in the same Namespace)
+- Guidance to operator authors to structure controller code to alert on missing cloud credentials, not crash.
+
+SDK to support this new template. SDK to validate and in particular: alert on any permission changes between operator
+versions.
+
+### Workflow Description
+
+Making a layered operator ready to work on an STS enabled cluster will involve the following steps:
+
+Operator installation under the proposed system:
+1. user goes to console and starts operator install
+2. console detects token-based (STS) cluster, and token-based-supporting operator
+3. console prompts user to create roles+supply ARN-type string
+4. console creates subscription with ARN-type string embedded as a spec.config.env
+5. operator deployment is created with ARN-type string in env
+6. operator creates `CredentialsRequest` including the ARN-type string
+7. Cloud Credential Operator populates Secret based on `CredentialsRequest`
+8. operator loads Secret, makes cloud service requests
+
+
+For the operator author team:
+- Add CredentialRequests to the bundle, known location;
+- Add "role-arn"-type fields to the CredentialRequests
+- Add references to the CredentialRequests in the CRD spec;
+- Use SDK to validate bundle to catch permission changes or mis-configuration
```

Comment:
Is this some sort of static validation of the annotations? Or is this suggesting that SDK somehow assesses the requests the operator intends to make and warns/errors if they require increased permission?